src/main/java/by/onliner/utils/Recorder.java

```diff
@@ -0,0 +1,120 @@
+package by.onliner.utils;
+
+import org.monte.media.Format;
+import org.monte.media.Registry;
+import org.monte.media.math.Rational;
+import org.monte.screenrecorder.ScreenRecorder;
+import ws.schild.jave.*;
+import java.awt.*;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import static org.monte.media.FormatKeys.*;
+import static org.monte.media.VideoFormatKeys.*;
+
+/***
+ * The class is used to record screen and convert video to mp4 format.
+ */
+public class Recorder extends ScreenRecorder {
+
+    public static ScreenRecorder screenRecorder;
+    private static String name;
+    public static final String savePath = "./Videos/";
+
+    /***
+     * Setting video recorder
+     * @param cfg graphics configuration
+     * @param captureArea capture size
+     * @param fileFormat format of the file
+     * @param screenFormat format of the screen
+     * @param mouseFormat format of the mouse
+     * @param audioFormat audio format
+     * @param movieFolder movie file
+     * @param name method name
+     */
+    public Recorder(GraphicsConfiguration cfg, Rectangle captureArea, Format fileFormat,
+                    Format screenFormat, Format mouseFormat, Format audioFormat, File movieFolder, String name)
+            throws IOException, AWTException {
+        super(cfg, captureArea, fileFormat, screenFormat, mouseFormat, audioFormat, movieFolder);
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH.mm.ss");
+        Recorder.name = name + "-" + dateFormat.format(new Date()) + "." + Registry.getInstance().getExtension(fileFormat);
+    }
+
+    /***
+     * Creates a file for recording the movie.
+     * @param fileFormat format of the file
+     * @return created file
+     */
+    @Override
+    protected File createMovieFile(Format fileFormat) throws IOException {
+
+        if (!movieFolder.exists()) {
+            movieFolder.mkdirs();
+        } else if (!movieFolder.isDirectory()) {
+            throw new IOException("\"" + movieFolder + "\" is not a directory.");
+        }
+        return new File(movieFolder,
+                Recorder.name);
+    }
+
+    /***
+     * Initialise recorder and start recording test
+     * @param name method name
+     */
+    public static void startRecording(String name) throws Exception {
```

Comment:
Do you need to pass video file name to both start & stop methods?

Reply:
Yes, if i need the clear file name i need to pass video file name to both start & stop methods

Reply:
Is that possible to update that? why would you need it for both start&stop?

Reply:
To give video name like method name